x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -726,6 +782,31 @@ export class AlertsClient {
     }
   }
 
+  public async bulkUpdateCases({ ids, index, caseIds }: BulkUpdateCasesOptions) {
+    /**
+     * We do this check to avoid any mget calls or authorization checks.
+     * The check below does not ensure that an alert may exceed the limit.
+     * We need to also throw in case alert.caseIds + caseIds > MAX_CASES_PER_ALERT.
```

Comment:
I assume the auth check that users have write access to cases is occurring in the cases client before this is called?

Reply:
Correct. The cases RBAC will throw an error if you do not have access to update the case. Our RBAC integration tests cover this scenario. Though if a plugin outside of cases uses the alerts client to update the alert's data it will bypass the auth check. Though, plugins should use the cases' client to attach alerts to a case and not use the alerts' client directly. I will add a comment in the code to emphasize this.